src/errors.rs

```diff
@@ -9,49 +9,6 @@
 
 //! Errors which may occur when parsing keys and/or signatures to or from wire formats.
```

Comment:
This is perhaps the most invasive change in the current implementation: it eliminates the `InternalError` type entirely, and uses the opaque `signature::Error` type as `SignatureError`.

It's still possible to support `InternalError` using the current `signature` crate via its `Error::source` feature, but only when `std` is enabled:

https://docs.rs/signature/1.0.0-pre.1/signature/struct.Error.html#method.from_source

However, as noted in the docs above, the `signature` crate discourages this to avoid sidechannel leakage that might allow the attacker to either recover a private key or forge a signature.

If you'd prefer to keep the `InternalError` type, I can use `Error::from_source` to capture it, and expose it as `Error::source`, but only when the `std` feature is enabled (since it uses `Box` to store it internally, and exposes it through the `std::error::Error` trait)